### DISCUSSION TITLE
tutorials/lift-off-part1/02-feature-data-requirements

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#47] [course feedback]
# tutorials/lift-off-part1/02-feature-data-requirements

https://www.apollographql.com/tutorials/lift-off-part1/02-feature-data-requirements

**Comment by mathoks** [comment]
Thanks I'm following

**Comment by vasily-mishanin** [comment]
Hope it will helps me in new project and make solid skills

**Comment by monicara67** [comment]
So far, so good!

**Comment by bhagatabhijeet** [comment]
Fantastico!  loving it 💗

**Comment by Tritintruong114** [comment]
well document !

**Comment by saiyantanghosh** [comment]
following

**Comment by marksikaundi** [comment]
Being new to graphql this is one of the best as a kick start for me.

**Comment by repiemm** [comment]
So each node "property" can be a relationship with another node, forming edges... Got it!

**Comment by lycha0206** [comment]
for that exercise, a graph is not a database because? It is true that a graph represents what we want to get from a portion of the database: the node and how it joins another node, meaning how one table joins one table. So it is a portion of a DB, not the DB itself.

**Reply by lizhennessy** [reply]
Thanks for the feedback @lycha0206! In this assessment, we're primarily looking to distinguish our data graph in GraphQL from the concept of an actual database, such as SQL or MongoDB. The concepts behind GraphQL do indeed have a lot in common with how databases are put together - nodes and edges, as you pointed out - however we don't use our GraphQL graph to _store_ data. Instead, it helps us describe the relationships between different data types, and provides an intuitive interface for anyone who's looking to query that data.

**Comment by lycha0206** [comment]
the picture showed an arrow go outside to the "title"... which is not a node, but just value. How is drawing it this way, vs if I put all the values right inside the node itself.  Kind of like, a node is like an object, with properties either as a value, or as a foreign key to identify another record in another table?

**Reply by lizhennessy** [reply]
Hey @lycha0206 ! The illustration you're referring to shows us how we can break down our data into a system of types and fields. In this case, the "node" is the type that we define in GraphQL. Each arrow - whether it points to another type, or simply a value - represents one of the type's _fields_. Each of these fields tell us something we can "ask about", or query for. You'll see how this comes to life in Schema Definition Language (SDL) in the upcoming lesson of the course!

**Comment by NishantCoder108** [comment]
Feature Data Requirements ✅
--------------------------------

##  Important points i learn :
 -  Every object is node and its every relation is edges e.g.  we have card in which they have thumbnail , author image , author name, module count , length . so card is node and its connection its edges. Author is node and its details connection will be edges like author id . 
- Making schema ,it reduces the development  time and we work parallel .

**Comment by gustvoassuncao** [comment]
So far, so good!

**Comment by bocarw121** [comment]
Smooth so far, thanks!

**Comment by ACISSOKHO87** [comment]
The courses that i'm looking foor , thanks !

**Comment by emilmathew** [comment]
I am seeing this blank page (I can see 
Catstronaut
Kitty space academy icon and 2025 ©Apollo but in the middle nothing), any one got the similar problem? I do not seeing any title cards..

Server side:
[server] npm start                                                                                                                                                                                 main  ✭

> catstronauts-server-complete@1.0.0 start
> nodemon src/index

[nodemon] 3.0.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,cjs,json
[nodemon] starting `node src/index.js`
[nodemon] clean exit - waiting for changes before restart

Client Side:
[server] npm start                                                                                                                                                                                 main  ✭

> catstronauts-server-complete@1.0.0 start
> nodemon src/index

[nodemon] 3.0.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,cjs,json
[nodemon] starting `node src/index.js`
[nodemon] clean exit - waiting for changes before restart

**Reply by emilmathew** [reply]
·> node -v
v23.10.0

·> npm -v
10.9.2

**Reply by lizhennessy** [reply]
Hi there @emilmathew ! At this point, that's exactly what you should see. As you continue on through the course, you'll learn Schema Definition Language, build out the schema with types and fields, and get those cards rendering on the home page! (In the next part of the series, we'll move on from mocked data, so stay tuned!) Do let us know if you run into any other issues! :)

**Reply by emilmathew** [reply]
Ah, that's such a relief! Thanks for your help—I'll continue with this!

**Comment by omooladev** [comment]
Thank you. I've been working with GraphQL for a while, but revisiting the basics through this tutorial has really reignited my understanding. It's finally clicking in a whole new way
